compass/ocean/rpe.py

```diff
@@ -0,0 +1,126 @@
+        Name of the netCDF file containing the initial state
+
+    output_file_prefix : string
+        Prefix for the netCDF file containing output of forward step
+
+    num_files : int
+        Number of files on which to compute rpe
+
+    Returns
+    -------
+    rpe : numpy array of size (num_timesteps) x (num_files)
+    """
+    gravity = 9.80616
+
+    # --- Open and read vars from netCDF file
+    dsInit = xarray.open_dataset(initial_state_file_name)
+    nCells = dsInit.sizes['nCells']
+    nEdges = dsInit.sizes['nEdges']
+    nVertLevels = dsInit.sizes['nVertLevels']
+
+    xCell = dsInit.xCell
+    yCell = dsInit.yCell
+    xEdge = dsInit.xEdge
+    yEdge = dsInit.yEdge
+    areaCell = dsInit.areaCell
+    minLevelCell = dsInit.minLevelCell - 1
+    maxLevelCell = dsInit.maxLevelCell - 1
+    bottomDepth = dsInit.bottomDepth
+
+    # --- Compute a few quantities relevant to domain geometry
+    areaCellMatrix = np.tile(areaCell, (nVertLevels, 1)).transpose()
+    bottomMax = np.max(bottomDepth.values)
+    yMin = np.min(yEdge.values)
+    yMax = np.max(yEdge.values)
+    xMin = np.min(xEdge.values)
+    xMax = np.max(xEdge.values)
+    areaDomain = (yMax - yMin) * (xMax - xMin)
+
+    vert_index = \
+        xarray.DataArray.from_dict({'dims': ('nVertLevels',),
+                                    'data': np.arange(nVertLevels)})
+
+    cell_mask = np.logical_and(vert_index >= minLevelCell,
+                               vert_index <= maxLevelCell)
+    cell_mask = np.swapaxes(cell_mask, 0, 1)
+    nCells_1D = len(areaCellMatrix[cell_mask])
+
+    # --- Get number of timesteps, assuming all output files have the same
+    # --- number of timesteps
+    ds = xarray.open_dataset('{}{}.nc'.format(
+                             output_file_prefix, 1))
+    nt = ds.sizes['Time']
+    ds.close()
+
+    # --- Allocations
+    rpe1 = np.zeros(nCells_1D)
+    zMid = np.zeros(nCells_1D)
+    thickness = np.zeros(nCells_1D)
+    rpe = np.ones((num_files, nt))
+
+    for n in range(num_files):
+
+        ds = xarray.open_dataset('{}{}.nc'.format(
+                                 output_file_prefix, n+1))
+
+        xtime = ds.xtime.values
+        hFull = ds.layerThickness
+        densityFull = ds.density
+
+        for tidx, t in enumerate(xtime):
+
+            h = hFull[tidx, :, :].values
+            vol = np.multiply(h, areaCellMatrix)
+            density = densityFull[tidx, :, :].values
+            density_1D = density[cell_mask]
+            vol_1D = vol[cell_mask]
+
+            # --- Density sorting in ascending order
+            sorted_ind = np.argsort(density_1D)
+            density_sorted = density_1D[sorted_ind]
+            vol_sorted = vol_1D[sorted_ind]
+
+            thickness = np.divide(vol_sorted.tolist(), areaDomain)
+
+            # --- RPE computation
+            z = 0.0
+            for i in range(nCells_1D):
+                zMid[i] = z-thickness[i]/2.0
+                z = z-thickness[i]
```

Comment:
This is the kind of loop that one usually wants to avoid in python. It looks like the following might work instead:
```
z = np.append([0], -np.cumsum(thickness))
zMid = z[1:] - z[0:-1]
```